source/js/components/petition/petition.jsx

```diff
@@ -443,24 +443,29 @@ export default class Petition extends React.Component {
         <div>
           <p>{this.props.thankYou}</p>
           <div className="share-btn-container">
-            <button className="share-progress-btn share-progress-fb" onClick={e => this.fbButtonClicked(e)}>
-              <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 264 512">
-                <path fill="currentColor" d="M76.7 512V283H0v-91h76.7v-71.7C76.7 42.4 124.3 0 193.8 0c33.3 0 61.9 2.5 70.2 3.6V85h-48.2c-37.8 0-45.1 18-45.1 44.3V192H256l-11.7 91h-73.6v229"></path>
-              </svg>
-              SHARE
-            </button>
-            <button className="share-progress-btn share-progress-tw" onClick={e => this.twButtonClicked(e)}>
-              <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
-                <path fill="currentColor" d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"></path>
-              </svg>
-              TWEET
-            </button>
-            <button className="share-progress-btn share-progress-em" onClick={e => this.emButtonClicked(e)}>
-              <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
-                <path fill="currentColor" d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"></path>
-              </svg>
-              EMAIL
-            </button>
+            {this.props.shareLink ?
```

Comment:
This code will never execute. The block on line 435 comes after a check for the same variable, so if `this.props.shareLink` exists, line 435 will be returned and skip the remainder of this function.

There should be multiple checks for variables to resolve this issue- one for facebook, one for twitter, and one for email. The variable you're looking for is the ones defined [here](https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/wagtailpages/models.py#L300-L306). You're going to have to inspect the code to figure out how they eventually get passed to the client.

Reply:
@alanmoo  I got it.. and when I inspected the code there was no prop for sharelink_twiiter, facebook and email ... can you plz help me how can I get these fields from models to jsx as I am new to react... thanks ...

Reply:
Unfortunately if I had the time to trace that, I would be solving this ticket myself. If you can't figure it out, I'd suggest checking out our [good first issues](https://github.com/mozilla/foundation.mozilla.org/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)